wormgas/cogs/rainwave.py

```diff
@@ -885,6 +905,44 @@ async def vote(self, ctx: cmds.Context, candidate: int):
         else:
             await ctx.author.send('Your attempt to vote was not successful.')
 
+    @cmds.Cog.listener()
+    async def on_member_update(self, before: discord.Member, after: discord.Member):
+        if self.config.get("rainwave:donor_role_id"):
+            donor_role = self.config.get("rainwave:donor_role_id")
+            was_donor = donor_role in before.roles
+            is_donor = donor_role in after.roles
+            if is_donor and not was_donor:
+                await self.rw_enable_perks([after.id])
+        if before.nick != after.nick:
+            await self.rw_update_nickname(after.id, after.nick)
+
+    @cmds.Cog.listener()
+    async def on_user_update(self, before: discord.User, after: discord.User):
+        if before.avatar != after.avatar:
+            await self.rw_update_avatar(after.id, after.avatar)
+
+    async def _sync_donors(self):
+        if not self.config.get("rainwave:donor_role_id"):
+            return
+
+        donor_role = self.config.get("rainwave:donor_role_id")
+        donors: List[str] = []
+        for guild in self.bot.guilds:
+            for member in guild.members:
+                if donor_role in member.roles:
+                    donors.append(member.id)
+        await self.rw_enable_perks(donors)
+
+    @cmds.command()
+    @cmds.is_owner()
+    async def sync_donors(self):
```

Comment:
`discord.errors.ClientException: Callback for sync_donors command is missing "ctx" parameter.`

```suggestion
    async def sync_donors(self, ctx: cmds.Context):
```